### ISSUE TITLE
Wrong Interpretation of 32-bit Program Headers

### ISSUE THREAD

**Issue by blitz** [vincenthouyi/elf_rs#4]
The library currently mishandles 32-bit program headers. It's straight-forward to see when running the provided example with any 32-bit binary:

```
cargo run --example readelf -- some.elf32
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `target/debug/examples/readelf ../particle/pit_guesttest-elf32`
...
Program Header { type: LOAD, flags: 1000, offset: 800000, vaddr: 800000, paddr: 141d5, filesize: 141d5, memsize: 1, alignment: 1000 }
Program Header { type: LOAD, flags: 16000, offset: 815000, vaddr: 815000, paddr: 1208, filesize: 1208, memsize: 4, alignment: 1000 }
Program Header { type: LOAD, flags: 18000, offset: 817000, vaddr: 817000, paddr: 7c50, filesize: 20c118, memsize: 6, alignment: 1000 }
```

`memsize` and `flags` are obviously wrong. The reason is this code:

```Rust
#[derive(Debug)]
#[repr(C)]
pub struct ProgramHeaderGen<T> {
    p_type: u32,
    p_flags: u32,
    p_offset: T,
    p_vaddr: T,
    p_paddr: T,
    p_filesz: T,
    p_memsz: T,
    p_align: T,
}
```

The problem is that the flags field is _not_ at the same location for 32-bit and 64-bit ELFs. Given the design of the library, I also didn't find a straightforward fix. :(

See also https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#Program_header

**Comment by vincenthouyi** [comment]
Hi blitz, thank you for using this lib! Unfortunately the parsing for ELF32 program header is wrong. I'm working on a refactoring of the library including the fix for this issue. If your are looking for a elf parsing crate in no-std env, you may want to use  [xmas-elf](https://crates.io/crates/xmas-elf).

**Comment by vincenthouyi** [comment]
fixed in [v0.2.0](https://github.com/vincenthouyi/elf_rs/releases/tag/v0.2.0)
